picard/formats/util.py

```diff
@@ -45,6 +45,10 @@ def supported_formats():
     return [(file_format.EXTENSIONS, file_format.NAME) for file_format in _formats]
 
 
+def formats_with_sanitize_date():
+    return [f"{fmt.EXTENSIONS, fmt.NAME}" for fmt in _formats if hasattr(fmt, 'sanitize_date')]
```

Comment:
It should return `fmt` classes only and we could make it a generator:

```python
def formats_with_sanitize_date():
    for fmt in _format:
        if hasattr(fmt, 'sanitize_date'):
            yield fmt
```

To get the sorted list of names:

```python
fmt_names =  sorted(fmt.NAME for fmt in formats_with_sanitize_date())
```